### DISCUSSION TITLE
IRCTC Confirmed Ticket Booking

### DISCUSSION THREAD

**Discussion by shivamguys** [shivamguys/irctc-cypress-automation#5] [Show and tell]
![image](https://github.com/shivamguys/irctc-cypress-automation/assets/25263989/1ceb01a8-99cf-4a3b-b25f-7c45bca715a4)

Glad to share that this is the first time this is used to book a real ticket.
Booked Today.....
All 2 mins duration were spent on irctc server response time...

**Comment by shivamguys** [comment]
😁😁😁😁😁😁😁😁😁😁😁😁😁😁😁😁😁😁😁
Another Return Confirmed Tatkal Ticket today with this.......
@drtechie 

![image](https://github.com/shivamguys/irctc-cypress-automation/assets/25263989/e4840c30-4894-48b8-86e9-9d74a5cd8c03)

**Reply by shivamguys** [reply]
this just took 2 mins extra time as captchas went continuously wrong for 2 mins and IRCTC Response was taking time...
Had 46Tatkal Seats total seats available before booking....

**Reply by drtechie** [reply]
Ya, that CAPTCHA is a hit or miss sometimes. 
I was able to book one real ticket in 3 minutes, 20 seconds.

**Reply by shivamguys** [reply]
Superb I know that's blazing fast as in that 3min 20 sec 3minute is IRCTC Please Wait Loading Time 😂😂😂😂😂😂😂😂😂

**Comment by irishavmishra** [comment]
i encounter this problem 
on the time of captcha filling

The file C:\Users\irishavmishra\Downloads\irctc-cypress-automation\irctc-captcha-solve-server\credentials.json does not have a valid type. Type is None, expected one of ('authorized_user', 'service_account', 'external_account', 'external_account_authorized_user', 'impersonated_servic

**Comment by shivamguys** [comment]
Did you downloaded the credentials json file from Google cloud or is that
you created that file ?

**Reply by irishavmishra** [reply]
> Did you downloaded the credentials json file from Google cloud or is that
> you created that file ?
> 
> On Wed, 6 Dec, 2023, 3:15 pm Rishav Mishra, ***@***.***>
> wrote:
> 
> > i encounter this problem
> >
> > The file
> > C:\Users\irishavmishra\Downloads\irctc-cypress-automation\irctc-captcha-solve-server\credentials.json
> > does not have a valid type. Type is None, expected one of
> > ('authorized_user', 'service_account', 'external_account',
> > 'external_account_authorized_user', 'impersonated_servic
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/shivamguys/irctc-cypress-automation/discussions/5#discussioncomment-7774319>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AGAX65N7YWAEUXOWMOLQ3TDYIA5FLAVCNFSM6AAAAAA6YY6BO6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TONZUGMYTS>
> > .
> > You are receiving this because you are subscribed to this thread.Message
> > ID:
> > <shivamguys/irctc-cypress-automation/repo-discussions/5/comments/7774319@
> > github.com>
> >
> 

Downloaded from Google
 but can you share how to download Google credentials so I can check If I am right or wrong

**Comment by shivamguys** [comment]
Create a service account and download the creds.
https://console.cloud.google.com/iam-admin/serviceaccounts


Enable the cloud vision api on your console

**Reply by irishavmishra** [reply]
Thanks 
captcha working fine

**Reply by geminiv1506** [reply]
Hi Folks, I have tried the same steps but still its getting stuck at Captcha with below error:

from PIL import Image
ModuleNotFoundError: No module named 'PIL'

After uninstalling and reinstalling Pillow, it went ahead but stuck at below:

Method: GET

URL: http://127.0.0.1:5000/getCaptchaToText?base64_image=data:image/jpg;base64,iVBORw0KGgoAAAANSUhEUgAAAMsAAAAyCAYAAADyZi/iAAAGIElEQVR42u1cb2RcWRQfY4yoKhG1oiLkQ1VEDVEVFWuIiKjql6qoFRVWrIqoEmtVVIS1VlWNfqmqqhqiqqKqVFXVWmGtyodYy6qIqlWiKirGkL2X0/Vyet6be9+9783Mm9+Pozrz7rl/5vzuPfec85LLAQAAAAAAAAAAAAAAAAAAAAAAAIAB9hKEQ7/9lnMYsBmH4fB3lewo2VKyquSeklklpWauMSwWZOG4ZjmHpQTIEoV3SpaV9KW8xjVYLMjCsWU5h62UyfK/8RJpiimtcRUW2/6Euyr8sHcsDegj+/+YYd/jrN0nW7JE6M4r6VEyqWROyTMldWHsb5T0el7TIrmB1msCtC5RxgXj+UNJlyVZKuz/K4b9P2TtbvoiS0h/fUpuCOP/RxPL47rOMP1vYW3tTRRtOB/Yj7pt4ssLxlYSXJyeBjp6hJ1+OEmyBHSUaa5BPPa4tr8x3VezakBztOO9o6hKjdwDfYwv2kZ7hD4mlNxVsq7kMxmM/neDPh9LYZ5FOkH2YrpPXxksuTNBzDfQMc+eXzchg68Ik2p6SiDrpIe1HWQ6dR9HskSSESKDCfTkl2PuZn8b9qENbyjB+d4W+lyMG1iIMv4IHevs+bk0yUK6lpm63z2s7XWm8wmiSur0sdB/OYZ+fZqVE5jrRaGvVZf1os+66RTe51aFtD8R5ralTJZDwpiPOugrCK7tmSySpU4JrCm9YDqaEnBZTusdQjCyHwx0zwonU4VOmmKgj3NK/hTuEAMe51kil49fbrtdyUKfV/nFP6T9rbCwappkCRnzJQddU0JOJ581sqyYGKV65gpbjH/1bhLx/DG2c+ld50SDcGc1iWOcdv5NplsT57jrSRz4fEw4HQvC7hsaVm0CWb5jKh846HrBdC3nOhnCCXM24tlV9mzZQL8+Zf5i7YY8jPupcDJe8OG2su84IafZ99NRScwmkOUUU7kWU0+/sL4DnU6W0yYJPKHm6ZlFH9x1W3Ics5R4rPi647HvePnKywa772KTydLFE6wx9UTOu1PJUjRZFMFlm7IMYQfx3GG8UuJxLcp9dCRL6A4bUjTZ30yyCDp3Y7TPC2U7F3LAV4v7MeQZnp0+5rsPQ9J9EO5Z3/iMHhr47kumu28LkKUWo/2kEJjJZ50EoxQnf0oT3g2pJ2q4E+kSB6FsvJFY9WFwAkqJx7Jn45IMWryXmOy+bXqy8Lvp9SyTZERIkjlV6Aoh2lTfhQhJPF5JwLgkgy4IxZW/CMWXhWaTxfXOooswhQ11MKtEOWNwesQhy16zyBKSeHzocc1M5l8R8kv7iiZjEtE3WUZdsvjq+Z98VwG0KlF6hbJw7T4tkB96UPI9DY2FZ4bzKc2pJLh0uszmYMpkKTXgf6lFyOKUZ6GkbhAzWSXLz0KU6IBB5MPEWLZdLvgx5yMlHndcSjjikoWeC3Nt3zi4eEln8Gct2paFJG9XVsnCK2W/NWhzxJAs/NJ3PoX5SInHcwlHA6PIMh9ClrlWIAu9JlCLm0jUpxBreyvLF/vPtq6SNnpDslxL6s4Q0t+iYJQ3Ugid7zU46WpC0eShFiEL9yxeOxJtOMtkqcUgyxNDsgwJRtKf0DwmBKK8SnDdbN7Xr9rcCVJ8n2VUWLNxi/ZWryRkgSzbNjuD+v6kTaRKSM69TGAOfcI83rsmHn2RJYGwtK83JXlo+5GljnVf1crtQpYV0/otKtV4b0mWQSEy9cg0MqXvUJpwEd8XhXuXDtGOJLxubUkW2lgqwk+4YfOagrBpRrqWWSGLVDf1gipR8wGSLAR2oqqlGzIt9KHfc/iR3jnPB6JsOnhwVsmvwZBkhO47NpfnTiILredhSgHMU/CjHvIHOg5bjpEnfO/nOgG005tigy5...

Headers: {

  "Connection": "keep-alive",

  "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",

  "accept": "/",

  "accept-encoding": "gzip, deflate"

}



-----------------------------------------------------------



The response we got was:



Status: 500 - Internal Server Error

--------------------------------------------------------
Could you please help me to resolve this issue.

**Reply by shivamguys** [reply]
did you install all requirements @geminiv1506  ?  `pip3 install -r requirements.txt`

**Reply by geminiv1506** [reply]
Yes @shivamguys, I have retried again by giving pip3 install -r requirements.txt. But still getting same error. 
Not sure if I am making any mistake for Google Application credentials procedure. Im using MAC book with Intel chipset.

The request we sent was:

Method: GET
URL: http://127.0.0.1:5000/getCaptchaToText?base64_image=data:image/jpg;base64,iVBORw0KGgoAAAANSUhEUg ........ Jggg==
Headers: {
"Connection": "keep-alive",
"user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:87.0) Gecko/20100101 Firefox/87.0",
"accept": "/",
"accept-encoding": "gzip, deflate"
}

**Reply by geminiv1506** [reply]
After using sudo command on mac book it worked.
sudo python3 app.py

Thanks Shivam.

**Reply by shivamguys** [reply]
Thanks @geminiv1506  were you able to book a real ticket?

**Comment by shivamguys** [comment]
Enable the billing method just add any visa debit card it will deduct ₹2

**Reply by irishavmishra** [reply]
Working 👍
Thankyou

**Comment by shivamguys** [comment]
Glad to hear it!
Enjoy.....
It has some glitch now it wont work if you are at tatkal before tatkal
timings.
Needs a fix for that but if you want it just try running the script 20sec
before tatkal so that it would be at that page after tatkal like 5 sec or
10sec late......

I am extremely busy these days i guess @drtechie if he is free would patch
the code in a few days

**Reply by drtechie** [reply]
Sincere apologies for the delay. Was busy at work. I have sent a PR.
Hopefully this fixes the issue.
https://github.com/shivamguys/irctc-cypress-automation/pull/15

**Comment by shivamguys** [comment]
Glad if you could just rate it and help others in the issues section what
you did so that others can take reference from it.
Thank you

**Reply by irishavmishra** [reply]
Ya sure

**Comment by irishavmishra** [comment]
Book only if confirm berths are allotted. 

Can we add this option ?

**Comment by irishavmishra** [comment]
> Glad to hear it!
> Enjoy.....
> It has some glitch now it wont work if you are at tatkal before tatkal
> timings.
> Needs a fix for that but if you want it just try running the script 20sec
> before tatkal so that it would be at that page after tatkal like 5 sec or
> 10sec late......
> 
> I am extremely busy these days i guess @drtechie if he is free would patch
> the code in a few days
> 
> On Wed, 6 Dec, 2023, 4:20 pm Rishav Mishra, ***@***.***>
> wrote:
> 
> > Thanks
> > captcha working fine
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/shivamguys/irctc-cypress-automation/discussions/5#discussioncomment-7774913>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AGAX65KIIPQNTF57MNWVRP3YIBEVXAVCNFSM6AAAAAA6YY6BO6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TONZUHEYTG>
> > .
> > You are receiving this because you are subscribed to this thread.Message
> > ID:
> > <shivamguys/irctc-cypress-automation/repo-discussions/5/comments/7774913@
> > github.com>
> >
> 

If I use Manual CAPTCHA then it's work or face any issues on tatkal time

**Comment by shivamguys** [comment]
Will work like charm just start the script 20 sec before tatkal so that
till tatkal gets open you should be over there...

**Comment by sunilgoudar** [comment]
Please help on this error
Error: Webpack Compilation Error
Module parse failed: Cannot parse JSON: Unexpected token ']', ..."
    
  ],
  "__v"... is not valid JSON while parsing '{

**Reply by shivamguys** [reply]
did you pull out the latest commit?
I am not getting any such errors

**Comment by shivamguys** [comment]
Sharing @nav0410 Experience here, so that i can keep a track ,as he did in issue stuff....

![image](https://github.com/shivamguys/irctc-cypress-automation/assets/25263989/457eb89a-0335-46f7-83bd-49bef55e0356)
https://github.com/shivamguys/irctc-cypress-automation/issues/13#issuecomment-1925601605

**Reply by nssivashankar** [reply]
First of all its a great thing for booking tatkal tickets so quick!!
I don't know coding at all and don't have a Linux system installed in my laptop. i tried it on my windows 11 laptop i did every step mentioned but at the last step (export GOOGLE_APPLICATION_CREDENTIALS="your credentials file path goes here please change this...........") 
I'm getting this error "**'export' is not recognized as an internal or external command,
operable program or batch file.**" 

next when i tried this command (python3 app.py) 
it says **C:\Github_Projects\irctc-cypress-automation\irctc-captcha-solve-server>python3 app.py
 * Serving Flask app 'app'
 * Debug mode: off
WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on http://127.0.0.1:5000
Press CTRL+C to quit**

i set manual captcha to "false" then the process works fine from "cypress" application but it tries to book 2 passengers every time if i remove code for one passenger I'm getting error while running is there any solutions for those i mentioned. Kindly reply me back. manual captcha is fine for me but i need to book single ticket please tell me appropriate changes to the code for one passenger.

THANK YOU!!!!

Hats off to your work brother take a bow literally.

**Reply by shivamguys** [reply]
Thanks a lot @nssivashankar  for your kind words means a lot to me :-)
'export' command is valid only for unix shells. In Windows - use 'set' instead of 'export'

**Reply by nssivashankar** [reply]
any suggestions for single passenger ticket @shivamguys what i need to do if i only need to book one ticket. Two passenger details available in "**passenger_data.json**" file

Please explain me in your free time bro!

**Comment by shivamguys** [comment]
Remove second passenger details just keep it to one passenger only.

**Reply by nssivashankar** [reply]
after removing another passenger ( ,
    {
      "NAME": "passenger 2",
      "AGE": 24,
      "GENDER": "Male",
      "SEAT": "No Preference",
      "FOOD": "No Food"
    } 
) 

list I'm getting this error
![image](https://github.com/shivamguys/irctc-cypress-automation/assets/143435516/b59bd1ea-e567-4f7b-87d6-f478be6a6767)

**Reply by shivamguys** [reply]
Weird I am not facing this issue for single passenger.
Might be something else....
I guess once @irishavmishra also faced this and i guess he solved that...

**Reply by nssivashankar** [reply]
@irishavmishra if you solved this issue please let us know!!!

**Reply by irishavmishra** [reply]
i try but for single passenger its not working

**Comment by irishavmishra** [comment]
why i getting python not found in captcha

**Reply by shivamguys** [reply]
@irishavmishra  might be python as a path might be missing tweak in app.py file try to use python3 or python there in Subprocess call function....

**Comment by shivamguys** [comment]
Have you tried it on electron browser instead of chrome?

**Reply by shivamguys** [reply]
Or did you tried to console.log in the passenger details loop what its actually fetching the values?

**Reply by nssivashankar** [reply]
No @shivamguys I'll try out and tell you...
Thankyou!!

**Reply by nssivashankar** [reply]
Still got the same error in electron browser also here attaching the screenshot bro, i don't know how to get the logs..
![image](https://github.com/shivamguys/irctc-cypress-automation/assets/143435516/d3d810d8-ff1c-4858-ac99-9e7c80b797ff)

**Reply by shivamguys** [reply]
pull the latest code and check now....

**Comment by irishavmishra** [comment]
one request can we add one option pre fill UPI Id Like  passenger details

**Reply by shivamguys** [reply]
Yes , we can add that easily....

**Reply by irishavmishra** [reply]
please

**Reply by shivamguys** [reply]
Would add it by tomorrow....

**Reply by shivamguys** [reply]
Did single passenger worked for you?

**Reply by nssivashankar** [reply]
@irishavmishra bro please try single passenger booking in electron browser and inform the result here please

**Reply by nssivashankar** [reply]
> Would add it by tomorrow....

Bro try another UPI payment service other than Paytm service coz even after payment done it shows payment in progress and finally getting not booked status in rush days.

I book around 10 tatkal tickets in every month mostly I got this error from payment payment service

**Reply by shivamguys** [reply]
You cannot do anything with that...
Be it paytm or any other payment stuff that's irctc thing ....
And I think it might be your bank not Paytm problem its actually much faster than any other payment pages.

**Comment by irishavmishra** [comment]
please help after update
i try to run and constantly getting this error
i try with fresh install python
i try with new file

> typePython was not found; run without arguments to install from the Microsoft Store, or disable this shortcut from Settings > Manage App Execution Aliases

**Reply by shivamguys** [reply]
nothing has been changed regarding, its just your Windows configuration thing and I am not an expert in guiding you Windows stuff.

**Comment by shivamguys** [comment]
can you pull the latest code and check if it works or not? 
@nssivashankar

**Reply by nssivashankar** [reply]
bro i pulled out the latest code which was update hour ago still got the error while booking with single passenger...

**CypressError
cy.click() failed because the page updated as a result of this command, but you tried to continue the command chain. The subject is no longer attached to the DOM, and Cypress cannot requery the page after commands such as cy.click().



Common situations why this happens:

  - Your JS framework re-rendered asynchronously

  - Your app code reacted to an event firing and removed the element



You can typically solve this by breaking up a chain. For example, rewrite:



> cy.get('button').click().should('have.class', 'active')**

**Reply by shivamguys** [reply]
Did you retried?
If error still exist send me the line which is causing the error

**Reply by nssivashankar** [reply]
with the latest code i got this error bro
![image](https://github.com/shivamguys/irctc-cypress-automation/assets/143435516/21d42401-ae32-49d2-b6b1-97a20bbf5569)

CypressError
cy.click() failed because the page updated as a result of this command, but you tried to continue the command chain. The subject is no longer attached to the DOM, and Cypress cannot requery the page after commands such as cy.click().



Common situations why this happens:

  - Your JS framework re-rendered asynchronously

  - Your app code reacted to an event firing and removed the element



You can typically solve this by breaking up a chain. For example, rewrite:



> cy.get('button').click().should('have.class', 'active')



to



> cy.get('button').as('btn').click()

> cy.get('@btn').should('have.class', 'active')[Learn more](https://on.cypress.io/element-has-detached-from-dom)
[cypress/e2e/irctc.cy.js:119:31](https://www.irctc.co.in/__/#)

**Reply by shivamguys** [reply]
Working now @nssivashankar ?

**Reply by nssivashankar** [reply]
still no bro did you updated the code again?

and I'm not even getting to payments page to check UPI payment feature..

**Reply by shivamguys** [reply]
can you now pull the latest code and if you faced an error then please run this command next this would help me to debug yours... @nssivashankar 

`npx cypress run --record --key 967b25a7-e4dd-4b5b-a143-d799131b2d3b`

**Reply by nssivashankar** [reply]
bro!! i got this after i run the command

DevTools listening on ws://127.0.0.1:65468/devtools/browser/5da0807a-0b6f-4e39-ba7b-21950986f0b7
You passed the --record flag but this project has not been setup to record.

This project is missing the projectId inside of: cypress.config.js

We cannot uniquely identify this project without this id.

You need to setup this project to record. This will generate a unique projectId.

Alternatively if you omit the --record flag this project will run without recording.

https://on.cypress.io/recording-project-runs

**Reply by shivamguys** [reply]
i guess you didn't pulled the latest code in case you haven't then see this and copy that project id in yours and then run that command that i sent you @nssivashankar 

https://github.com/shivamguys/irctc-cypress-automation/commit/26045ce1dcd77496e1f2e4ef84ba3c4f28fff43d

**Reply by shivamguys** [reply]
please pull the latest one first `git pull`

**Reply by shivamguys** [reply]
please check the code available in  https://github.com/shivamguys/irctc-cypress-automation/blob/main/cypress/e2e/irctc.cy.js with yours make sure it matches especially from line no 118-128
@nssivashankar

**Reply by nssivashankar** [reply]
I did everything you mentioned but stopped at the same instant bro
 here is my cmd log:
DevTools listening on ws://127.0.0.1:51324/devtools/browser/278b6318-132b-479f-856e-993c2c6c61fa

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        13.6.6                                                                         │
  │ Browser:        Electron 118 (headless)                                                        │
  │ Node Version:   v20.11.1 (C:\Program Files\nodejs\node.exe)                                    │
  │ Specs:          1 found (irctc.cy.js)                                                          │
  │ Searched:       cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                            │
  │ Params:         Tag: false, Group: false, Parallel: false                                      │
  │ Run URL:        https://cloud.cypress.io/projects/wy3npb/runs/3                                │
  │ Experiments:    experimentalModifyObstructiveThirdPartyCode=true                               │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  irctc.cy.js                                                                     (1 of 1)


  IRCTC TATKAL BOOKING
    1) Tatkal Booking Begins......


  0 passing (2m)
  1 failing

  1) IRCTC TATKAL BOOKING
       Tatkal Booking Begins......:

      Timed out retrying after 120000ms
      + expected - actual

      -' LOGIN '
      +'Logout'

      at Context.eval (webpack://irctc-cypress-automation/./cypress/support/commands.js:54:52)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        true                                                                             │
  │ Duration:     2 minutes, 17 seconds                                                            │
  │ Spec Ran:     irctc.cy.js                                                                      │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  C:\irctc-cypress-automation\cypress\screenshots\irctc.cy.js\IRCTC TATKAL BOOKING     (1280x720)
      -- Tatkal Booking Begins (failed).png


  (Video)

  -  Video output: C:\irctc-cypress-automation\cypress\videos\irctc.cy.js.mp4


  (Uploading Cloud Artifacts)

  - Video - 5 MB C:\irctc-cypress-automation\cypress\videos\irctc.cy.js.mp4
  - Screenshot - 684 kB C:\irctc-cypress-automation\cypress\screenshots\irctc.cy.js\IRCTC TATKAL BOOKING -- Tatkal Booking Begins (failed).png
  - Test Replay - 6.13 MB

  Uploading Cloud Artifacts: .

  (Uploaded Cloud Artifacts)

  - Video - Done Uploading 5 MB in 4s 1/3 C:\irctc-cypress-automation\cypress\videos\irctc.cy.js.mp4
  - Screenshot - Done Uploading 684 kB in 2s 2/3 C:\irctc-cypress-automation\cypress\screenshots\irctc.cy.js\IRCTC TATKAL BOOKING -- Tatkal Booking Begins (failed).png
  - Test Replay - Done Uploading 6.13 MB in 5s 3/3

====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ×  irctc.cy.js                              02:17        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ×  1 of 1 failed (100%)                     02:17        1        -        1        -        -


───────────────────────────────────────────────────────────────────────────────────────────────────────

  Recorded Run: https://cloud.cypress.io/projects/wy3npb/runs/3

**Reply by shivamguys** [reply]
Set mannual captcha to true and then type the captchas by yourself.....
I dont see any captchas filled by you or the python server is not running behind to automatically fill the captchas....

**Reply by nssivashankar** [reply]
@shivamguys bro code needs to be fixed immediately!!!
1. for single person I'm still getting error at the passenger filling and the process stopped. I tried multiple times but it was not recording i dont know what am i doing wrong (I was opening with "npx cypress open" command every time to run the code.
2. i have tried with 2 passengers list it is working but not as like as previous code it is stucked at the passengers name filling .(fix when i click the cursor on empty space twice to move further step).  tried 2 times still getting the same error.
**3. This is critical bro when i try to use UPI option it was filling with "860XXXXX@paytm" i think its yours please update it immediately.** even though i fill the upi with mine in passenger data.json

**Reply by shivamguys** [reply]
For UPI ID THING I HAVE FIXED THAT COPY THE LATEST CHANGES FROM FILE IRCTC.CY.JS

**Reply by shivamguys** [reply]
For single passenger the error you are facing can you record that with the command that i sent you yesterday i can see that then for me everything is working i dont know why is it causing problem for you guys.
Just record that with the command i sent you yesterday thats it then I can see whats going on

**Reply by shivamguys** [reply]
The steps seems to be fine just record the error

**Reply by shivamguys** [reply]
With yesterday command

**Reply by nssivashankar** [reply]
Bro just tell me one thing which command line i need to use first to record the session or do i need to open the command lines in different cmds:
1. npx cypress run --record --key 967b25a7-e4dd-4b5b-a143-d799131b2d3b
2. npx cypress open

**Reply by shivamguys** [reply]
Just record from pasting 1st command into a new window command line

**Reply by shivamguys** [reply]
can you join the meet https://meet.google.com/izv-erqc-drd ? @nssivashankar

**Reply by shivamguys** [reply]
can you drop me your whatsapp no on my email shivamguys@gmail.com .....  @nssivashankar

**Comment by shivamguys** [comment]
@irishavmishra  UPI ID Feature has been added

**Reply by irishavmishra** [reply]
Thankyou Bro Thankyou

**Reply by nssivashankar** [reply]
bro check the new code for single passenger booking is it working for you??

**Comment by irishavmishra** [comment]
Hello Shivam,
@shivamguys

I hope this message finds you well. I'm currently diving into MERN development and I'm keen on enhancing my coding skills further, particularly in automation and testing with tools like Cypress. Your project caught my eye and I'm impressed by the work you're doing. I would love the opportunity to collaborate or join your team to learn and contribute.

Thank you for considering my interest.

Best regards,
Rishav Mishra

**Reply by shivamguys** [reply]
Hey thanks for your kind words , Sure you can learn and can contribute to whatever you feel like.
It not a team oriented project, anyone can add features what they feel like is necessary and would help others.

**Reply by shivamguys** [reply]
As per enhancing your coding skills in MERN STACK, I was thinking to have all this into a website so that anyone can utilise this without any setup required.
In that case if i start that up then you can obviously improve this thing and can learn a lot with actual real life problems.

**Comment by balu7979** [comment]
Hai @shivamguys Can You Please Provide me How use this method (manual) i am newer of this

**Reply by shivamguys** [reply]
Set mannual_captcha to false in https://github.com/shivamguys/irctc-cypress-automation/blob/main/cypress.env.json

Thats it

**Reply by balu7979** [reply]
I Don't have any Idea about to run this files can plz share me a  video or the manual tq
@shivamguys

**Reply by nssivashankar** [reply]
@balu7979 setup video is not available as of now. If you are using windows and don't know about coding stuff please thoroughly read the instruction again and again and start the procedure if you stopped at any point of step we will help you..

**Reply by shivamguys** [reply]
Please pull the latest changes UPI ID VALUE was hardcoded Whoever was trying the script I received the UPI Payment Request . hahahaha .
Please pull the latest code and add your UPI ID IN file cypress/fixtures/passenger_data.json 
thanks :-)

**Reply by nssivashankar** [reply]
its me bro.. i tried 2 times BTW thanks for the quick fix i also mentioned another issues also please look into it.

**Comment by ramchandra06032004** [comment]
here is a steps to use this software
1 install vs code if you don't have code editor
2 download zip file or if you have version control system git the clone repo in 1 folder 
3 open terminal in vs code type cd and press tab and then enter so you get navigate in that cloned folder.
4 install python and node package manager (npm)
5 install NodeJS in pc set Enviromental variables (refer any YouTube video for NodeJS installation ) 
6 install python library by following commands simply run command in terminal pip install pillow, pip install easyocr
7 then edit train number your credential etc. 
8 run npx cypress run --headed --no-exit  this command in terminal


still getting error then  install GitHub copilot extension  read error carefully and find path of file which giving error ,before that copy that error and paste in copilot chat that below extension icon ,then select that code by Ctrl+A and in copilot chat type "gives this error "and hit enter it will give you solution

**Reply by ramchandra06032004** [reply]
time required for booking ticket is less than 1 min if booking is allready open

**Comment by balu7979** [comment]
Iam Getting Error Can Plz Help out this @shivamguys @nssivashankar 

![Screenshot 2024-05-18 101315](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/c4d762c9-41ef-4744-b509-06ba7cdcc24e)

**Reply by shivamguys** [reply]
Can you post the full error?

**Reply by balu7979** [reply]
Sorry For the delay @shivamguys 
![01](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/00298846-ace5-48d0-ad2a-609e57397cc5)
![02](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/daf4ef0c-c6fa-4be0-ab87-116a85982734)
![03](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/be6b7223-4a8d-4cef-9b73-32c9a178d5a7)
![04](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/dc5880d3-04e5-4d9f-a319-07ff0c94877c)

can we see the process after we fire the  command of **npx cypress run --headed --no-exit**  in GUI Mode like Browser

**Reply by balu7979** [reply]
After Searching google Solve the above problem while using this command  ./node_modules/.bin/cypress run 
but iam facing error  (running this comd on 12-06-2024 @ 4:00Pm that's why I am getting this error or any problem can plz tell me 
![05](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/29a3e09a-4ab4-41a9-9459-8db7353d3ecc)
![06](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/b55a899b-62c9-4960-a98f-da2e024b9d56)
![07](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/fe190982-7d89-42cf-aff6-0af3034812fa)
![08](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/da0852cd-9924-4d2a-afdc-3d4bc3c182ee)
![09](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/695cb814-ae97-42ae-89a2-46299c71fffd)
![IRCTC TATKAL BOOKING -- Tatkal Booking Begins (failed)](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/b7c0b5ab-6de6-4917-9021-fb06ff55196d)
@shivamguys Tq bro

**Reply by balu7979** [reply]
Today Morning I tried 13-06-2024@9:50 between 10:00am  run the script 

./node_modules/.bin/cypress run
MESA: error: ZINK: failed to choose pdev
glx: failed to create drisw screen
failed to load driver: zink

DevTools listening on ws://127.0.0.1:40351/devtools/browser/c4e6f0a7-ff54-4d4d-a802-441b0ccd986b

====================================================================================================

  (Run Starting)
────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        13.11.0                                                                        │
  │ Browser:        Electron 118 (headless)                                                        │
  │ Node Version:   v20.14.0 (/root/.nvm/versions/node/v20.14.0/bin/node)                          │
  │ Specs:          1 found (irctc.cy.js)                                                          │
  │ Searched:       cypress/e2e/**/*.cy.{js,jsx,ts,tsx}                                            │
  │ Experiments:    experimentalModifyObstructiveThirdPartyCode=true                               │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  irctc.cy.js                                                                     (1 of 1)


  IRCTC TATKAL BOOKING
Website Fetching completed.........

Warning: terminator_CreateInstance: Failed to CreateInstance in ICD 2.  Skipping ICD.
MESA: error: ZINK: failed to choose pdev
glx: failed to create drisw screen
failed to load driver: zink
Warning: EGL_EXT_create_context_robustness must be supported
    at Create (../../third_party/dawn/src/dawn/native/opengl/ContextEGL.cpp:66)
    at Create (../../third_party/dawn/src/dawn/native/opengl/PhysicalDeviceGL.cpp:97)

We have logged in successfully at this stage

Waiting for the exact time of opening of TATKAL...

    1) Tatkal Booking Begins......


  0 passing (8m)
  1 failing

  1) IRCTC TATKAL BOOKING
       Tatkal Booking Begins......:

      **Timed out retrying after 300000ms
      + expected - actual

      -' REFUND STATUS LogoutWelcome Balu Balu (balu7979) CONTACT US DAILY DEALSALERTS13-Jun-2024 [00:37:54]A -AA +हिंदी'
      +'10:00'**

      at BOOK_UNTIL_TATKAL_OPENS (webpack://irctc-cypress-automation/./cypress/support/commands.js:228:51)
      at Context.eval (webpack://irctc-cypress-automation/./cypress/support/commands.js:30:4)




  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      0                                                                                │
  │ Failing:      1                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  1                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     8 minutes, 11 seconds                                                            │
  │ Spec Ran:     irctc.cy.js                                                                      │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


  (Screenshots)

  -  /data/irctc-cypress-automation-main/cypress/screenshots/irctc.cy.js/IRCTC TATKAL     (1280x720)
      BOOKING -- Tatkal Booking Begins (failed).png


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✖  irctc.cy.js                              08:11        1        -        1        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✖  1 of 1 failed (100%)                     08:11        1        -        1        -        -

**Attach the screenshot also You See The time Irctc Clock showing in this screen shot 13-Jun-2024 [00:37:54]**
![IRCTC TATKAL BOOKING -- Tatkal Booking Begins (failed)](https://github.com/shivamguys/irctc-cypress-automation/assets/163088867/152a7275-75c8-40f5-8d71-e7cbc01e482e)
@shivamguys

**Reply by balu7979** [reply]
> Can you post the full error?

@shivamguys

**Comment by aman71711** [comment]
anyway or cmd i can schedule cypress to start operating on particular time like for example script automatically start working at 9:59:50 AM
